WordPress/src/test/java/org/wordpress/android/viewmodel/main/MySiteViewModelTest.kt

```diff
@@ -0,0 +1,204 @@
+package org.wordpress.android.viewmodel.main
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import android.arch.lifecycle.Observer
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mock
+import org.mockito.Mockito.never
+import org.mockito.Mockito.times
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.annotations.action.Action
+import org.wordpress.android.fluxc.model.PlanModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.SiteStore.OnPlansFetched
+import org.wordpress.android.fluxc.store.SiteStore.PlansError
+
+@RunWith(MockitoJUnitRunner::class)
+class MySiteViewModelTest {
+    @Rule @JvmField val rule = InstantTaskExecutorRule()
+
+    @Mock private lateinit var dispatcher: Dispatcher
+    @Mock private lateinit var store: SiteStore
+    private lateinit var viewModel: MySiteViewModel
+
+    @Mock private lateinit var siteObserver: Observer<SiteModel>
+
+    private val site = SiteModel().apply { id = 1 }
+
+    @Before
+    fun setUp() {
+        viewModel = MySiteViewModel(store, dispatcher)
+    }
+
+    @Test
+    fun currentPlanIsSetWhenPlansAreLoaded() {
+        val plans = listOf(
+                PlanModel(
+                        1,
+                        "slug",
+                        "name",
+                        isCurrentPlan = false,
+                        hasDomainCredit = false
+                ),
+                PlanModel(
+                        2,
+                        "slug",
+                        "name",
+                        isCurrentPlan = true,
+                        hasDomainCredit = true
+                )
+        )
+
+        whenever(dispatcher.dispatch(any())).then {
+            viewModel.onPlansFetched(OnPlansFetched(site, plans, null))
+        }
+
+        viewModel.currentPlan.observeForever { plan ->
```

Comment:
The only reason I used a mock for the siteObserver was because I wanted to assure that it wasn't called in the `setSiteWillNotFireObserversIfNewSiteHasSameIdAsOldSite` test. Using a mock was the only way I could think of to verify that. I try to use real objects in tests when possible because it is possible to test the wrong thing (or nothing at all) when you create a mock that doesn't act like the real object.